src/Backups/BackupImpl.cpp

```diff
@@ -372,40 +546,39 @@ void BackupImpl::readBackupMetadata()
     String str;
     readStringUntilEOF(str, *in);
     increaseUncompressedSize(str.size());
-    std::istringstream stream(str); // STYLE_CHECK_ALLOW_STD_STRING_STREAM
-    Poco::AutoPtr<Poco::Util::XMLConfiguration> config{new Poco::Util::XMLConfiguration()};
-    config->load(stream);
+    Poco::XML::DOMParser dom_parser;
+    XMLDocumentPtr config = dom_parser.parseMemory(str.data(), str.size());
```

Comment:
When I tried parsing that file, it got stuck in `Poco::Util::XMLConfiguration::load()`. I see you've changed that to `parseMemory()`. Does it work faster?

Reply:
When I played with that big file with 200K ```<file>``` elements I first saw this line being very slow https://github.com/ClickHouse/ClickHouse/pull/43627/files#diff-4848814683263bfb8784e5382bc1aff6e7075617c74451174b49f1819905374eL394
likely because of this https://github.com/pocoproject/poco/pull/3885 But even with the fix the loop that iterated over the files was slow because of many accesses by path like 'contents.file[N].name'. I suspect those might work in suboptimal way and take O(N), not O(1) or at least O(log(n)).
So I tried to use XMLDocument instead of XMLConfiguration and keep the pointer to the current file element. With this implementation the whole big file got processed in 1.5sec